lisa/sut_orchestrator/libvirt/platform.py

```diff
@@ -1341,3 +1378,73 @@ def _capture_libvirt_logs(self) -> None:
             )
             with open(str(libvirt_log_local_path), "w") as f:
                 f.write(libvirt_log)
+
+    def _add_device_passthrough(
+        self,
+        devices: ET.Element,
+        node_context: NodeContext,
+    ) -> ET.Element:
+        for config in node_context.device_passthrough_configs:
+            hostdev = ET.SubElement(devices, "hostdev")
+            hostdev.attrib["mode"] = "subsystem"
+
+            assert config.get("device_type")
+            hostdev.attrib["type"] = config.get("device_type", "")
+
+            assert config.get("managed_passthrough")
+            hostdev.attrib["managed"] = config.get("managed_passthrough", "")
+            if config.get("device_type") == "pci":
+                source = ET.SubElement(hostdev, "source")
+                src_addrs = ET.SubElement(source, "address")
+
+                source_domain = config.get("source_domain", "")
+                assert source_domain
+                src_addrs.attrib["domain"] = f"0x{source_domain}"
+
+                source_bus = config.get("source_bus", "")
+                assert source_bus
+                src_addrs.attrib["bus"] = f"0x{source_bus}"
+
+                source_slot = config.get("source_slot", "")
+                assert source_slot
+                src_addrs.attrib["slot"] = f"0x{source_slot}"
+
+                source_function = config.get("source_function", "")
+                assert source_function
+                src_addrs.attrib["function"] = f"0x{source_function}"
+
+                driver = ET.SubElement(hostdev, "driver")
+                driver.attrib["name"] = "vfio"
+        return devices
+
+    def _check_passthrough_device_exist(self, device_config: Dict[str, str]) -> bool:
```

Comment:
```suggestion
    def _device_exists(self, device_config: Dict[str, str]) -> bool:
```

This just checks if the specified device exists on the system, right? So, the name can be more concise.

Reply:
Done